app/javascript/gobierto_data/webapp/components/commons/Visualizations.vue

```diff
@@ -45,134 +44,91 @@ export default {
         this.checkIfQueryResultIsEmpty(newValue)
       }
     },
-    typeChart(newValue, oldValue) {
-      if (newValue !== oldValue) {
-        this.viewer.setAttribute('plugin', newValue)
-      }
-    },
-    arrayColumnsQuery(newValue, oldValue) {
-      if (JSON.stringify(newValue) !== JSON.stringify(oldValue)) {
-        this.viewer.clear()
-        this.viewer.update(this.items)
-        this.viewer.setAttribute('columns', JSON.stringify(newValue))
-      }
-    },
-    resetConfigViz(newValue) {
-      if (newValue) {
-        this.clearColumnPivots()
-      }
+    config() {
+      this.setConfig()
     }
   },
-  created() {
-    this.worker = perspective.worker();
-  },
   mounted() {
     this.viewer = this.$refs["perspective-viewer"];
     this.checkIfQueryResultIsEmpty(this.items)
+
+    this.viewer.addEventListener('perspective-config-update', this.handleConfigUpdates)
+  },
+  beforeDestroy() {
+    this.viewer.removeEventListener('perspective-config-update', this.handleConfigUpdates)
   },
   methods: {
     // You can run a query that gets an empty result, and this isn't an error. But if the result comes empty Perspective has no data to build the table, so console returns an error. We need to check if the result of the query is equal to the columns
     checkIfQueryResultIsEmpty(items) {
-      //The result of the query, if returns empty only contains the columns.
+      // The result of the query, if returns empty only contains the columns.
       const data = items.trim()
-      //Only the columns of the query.
-      const arrayColumnsQueryString = this.arrayColumnsQuery.toString()
+      // Only the columns of the query.
+      const [columns = ""] = items.split("\n");
+
+      // Clean any previous config
+      this.resetConfig();
+
+      this.viewer.setAttribute("columns", JSON.stringify(columns.split(",")))
+      if (this.configMap) {
+        const { metric, zoom = false } = this.configMap
+        this.viewer.setAttribute("metric", metric || "")
+        this.viewer.setAttribute("zoom", zoom || "")
+      }
```

Comment:
Event though we don't do nothing else yet, I think would be more appropriate set just one attribute, for instance, `config-map=""{\"zoom\":2,\"metric\":\"column\"}""`, i.e. a stringified object, in terms of a future enhancement where we need extra stuff.
Allowing an object from now on, we won't have such problem

Reply:
Done!

Reply:
That's done here. But is not in https://github.com/PopulateTools/perspective-map/blob/main/src/index.js. You will must adapt this attribute to be parsed from the plugin, and remove the old attributes